docs/webhost configuration sample.yaml

```diff
@@ -17,6 +17,10 @@
 # Web hosting port
 #PORT: 80
 
+# Ports used for game hosting. Values can be specific ports, port ranges or both. Default is: "49152-65535"
+# Examples of valid values: "40000-41000,49152-65535"
+#GAME_PORTS: "49152-65535"
```

Comment:
By documenting this in the config sample, it feels like we are now guaranteeing that all games will be hosted on ports within the given range, but currently that is still not the case in this PR.

If any Exception is raised during the `websockets.serve(...)` call on line 202 of `customserver.py`, the call is retried with port 0 which asks the OS for a random unused port and doesn't take into account this range.

Reply:
Yeah, it will do that. I was faced with a choice when working on this which was either
- keep the fallback for a random port if it fails for a soft-fail
- hard-fail it and prevent a server firing up when no port is used. 

I'm happy to document the current soft-fail nature of it on the sample config or push a commit that operates with a hard-fail, though my instinct was telling me a soft-fail was the safer approach to just getting servers up.

Reply:
Both of those are reasonable to me, I just would like it to be clearly documented if it's the soft-fail approach as a user might be confused if they specify a port and then get a port outside that range anyway. I know personally I was very confused when I looked in the code to find the range, opened up 49152-65535 on my firewall, and then got a port outside that anyway.

Maybe one of the core devs has an opinion on which one to do.

Hmm, I wonder if we could make it call `is_port_in_use` even when restarting an old room that has a `room.last_port`. If we did that then most common reason for the `websockets.serve` call to raise an Exception would go away.

Reply:
I feel the soft fail approach is better, though thinking about it more I might want to put a logging notice when it soft-fails so it's not obfuscated at all. 
I'll go ahead and also add some additional notes to the documentation on the soft-fail nature too

Reply:
My thought was also to add a loud `WARNING: Desired port 1234 is unavailable. Hosting on port 5678 instead.`

Reply:
I've gone ahead and made this message a fair bit loud and informative.